sdk/core/azure-core/src/exception.cpp

```diff
@@ -28,4 +30,48 @@ namespace Azure { namespace Core {
     RawResponse = std::move(rawResponse);
   }
 
+  RequestFailedException::RequestFailedException(
+      std::unique_ptr<Azure::Core::Http::RawResponse>& rawResponse)
+      : std::runtime_error(getRawResponseField(rawResponse, "message"))
+  {
+    auto& headers = rawResponse->GetHeaders();
+
+    StatusCode = rawResponse->GetStatusCode();
+    ErrorCode = getRawResponseField(rawResponse, "code");
+    StatusCode = rawResponse->GetStatusCode();
+    ReasonPhrase = rawResponse->GetReasonPhrase();
+    RequestId = HttpShared::GetHeaderOrEmptyString(headers, HttpShared::MsRequestId);
+    ClientRequestId = HttpShared::GetHeaderOrEmptyString(headers, HttpShared::MsClientRequestId);
+    Message = this->what();
+  }
+
+  std::string RequestFailedException::getRawResponseField(
```

Comment:
nit: This is an impl detail so not as important, but methods should be PascalCased, according to the guidelines.

Reply:
Fixed in #2536